src/bin.tsx

```diff
@@ -16,36 +19,61 @@ program
   .storeOptionsAsProperties(false)
   .passCommandToAction(false)
   .description('Deletes old GitLab pipelines')
-  .arguments('<gitlab-url>')
-  .arguments('<project-id>')
-  .arguments('<access-token>')
+  .arguments('[gitlab-url]')
+  .arguments('[project-id]')
+  .arguments('[access-token]')
   .option('-d --days <days>', 'older than days', '30')
   .option('--trace', 'show stack traces for errors when possible', false)
-  .action((gitlabUrl: string, projectId: string, accessToken: string, options: Record<string, unknown>) => {
-    const parsedProjectId = parseInt(projectId, 10);
-    const days = typeof options.days === 'string' ? parseInt(options.days, 10) : 30;
-    const startDate = new Date();
-    const listPipelinesFunction = listPipelines({ getRequest, gitlabUrl, projectId: parsedProjectId, accessToken });
-    const deletePipelineFunction = deletePipeline({
-      deleteRequest,
-      gitlabUrl,
-      projectId: parsedProjectId,
-      accessToken,
-    });
-    render(
-      <Main
-        gitlabUrl={gitlabUrl}
-        projectId={parsedProjectId}
-        accessToken={accessToken}
-        days={days}
-        startDate={startDate}
-        exit={exit}
-        listPipelines={listPipelinesFunction}
-        filterPipelinesByDate={filterPipelinesByDate}
-        deletePipeline={deletePipelineFunction}
-        showStackTraces={options.trace === true}
-      />,
-    );
-  });
+  .action(
+    async (
+      gitlabUrlArgument: string | undefined,
+      projectIdArgument: string | undefined,
+      accessTokenArgument: string | undefined,
+      options: Record<string, unknown>,
+    ) => {
+      const cliArguments: CliArguments = {
+        gitlabUrl: gitlabUrlArgument,
+        projectId: projectIdArgument ? parseInt(projectIdArgument, 10) : undefined,
+        accessToken: accessTokenArgument,
+        days: typeof options.days === 'string' ? parseInt(options.days, 10) : 30,
+        trace: options.trace === true,
+      };
+      const explorer = cosmiconfig('gitlab-pipeline-deleter');
+      const config = await loadConfig('./glpd.config.js', explorer);
+      const glpdArguments = mergeCliArgumentsWithConfig(cliArguments, config);
```

Comment:
So when I define e.g. `gitlabUrl` via CLI *and* config file, then the config file has precedence, right?

* I think this this should be documented
* Maybe it is better to prefer the CLI arguments in case you want to overwrite the config in the config file (e.g. useful in CI pipelines, where need to run the tool for an edge-case scenario with a different setting). WDYT?

Reply:
BTW: The precedence order should be also covered by unit tests.

Reply:
Never mind, seems like most of this is already implemented and tested. I was confused by the argument order here because other merge functions merge the arguments from right to left.

But I think it would be still good to document this behavior.

Reply:
It's the other way around. The CLI arguments are preferred over configuration file arguments. See also the unit test `mergeCliArgumentsWithConfig() prefers CLI arguments over configuration values`

I will update the documentation about this behaviour.

Reply:
> BTW: The precedence order should be also covered by unit tests.

It is in `mergeCliArgumentsWithConfig() prefers CLI arguments over configuration values`

Reply:
> other merge functions merge the arguments from right to left

True. I did this because of potential currying or composition where the last argument is always that one that is used for the function application.